code/game/machinery/recycler.dm

```diff
@@ -107,6 +107,8 @@
 		return
 	if(safety_mode)
 		return
+	if(iseffect(AM0))
+		return
```

Comment:
Same issue I have with #55382, this shouldn't need an iseffect() check on multipal objects, we want to manage this in one place

Reply:
Thoughts on how else I could do it? Because recycler indiscriminately eats anything that's not a mob, with some typechecked exceptions

Reply:
Perhaps have Destroy() require a force? Hm

Reply:
I'm hoping for more general fix that catches all the related issues with the system, rather then just throwing an iseffect() everywhere. I don't want some poor fuck to make a teleporter in 2 years that teleports lights. only issue is I'm not sure how to achieve this.

Reply:
An abstract flag would do it

Reply:
I believe this was discussed and it wasn't good enough, as everything would need an abstract check instead of the effect check, which still is 1 extra check.